pkg/cmd/issue/develop/develop.go

```diff
@@ -171,6 +171,14 @@ func developRunCreate(opts *DevelopOptions, apiClient *api.Client, issueRepo ghr
 		return err
 	}
 
+	// Remember which branch to target when creating a PR.
+	if opts.BaseBranch != "" {
+		err = opts.GitClient.SetBranchConfig(ctx.Background(), branchName, git.MergeBaseConfig, opts.BaseBranch)
```

Comment:
**Concern**

I'm a bit nervous about this purely because it's a breaking change.

Before this, development branches would always go back to the default branch unless otherwise specified with a flag. I don't have a clear answer for whether there are cases out there where people expect to create a development branch from a different branch than they want to PR back into.

Maybe we just bite the bullet and if someone complains, we can apologise. This definitely seems like the more natural default.

What do you think?

Reply:
I appreciate it's a change, but seems it's a change for good. The CLI created the topic branch from the `opts.BaseBranch` in the couple of code blocks above. If it were to target the default branch, the commit graph could be a real mess: the base branch may not be merged or, worse, has diverged from its base branch entirely e.g., a hotfix or feature branch, or just a branch for stabilization aka "release branch".

But, yes, it is a change. What if, as compensation for trying to do the, IMO, right thing here, interactive prompts to create a PR ask for the base (merge) branch with the current (or default) as the default value if they just hit Enter?